llama-index-integrations/indices/llama-index-indices-managed-vectara/llama_index/indices/managed/vectara/base.py

```diff
@@ -360,8 +363,14 @@ def as_query_engine(
             )
 
             retriever = self.as_retriever(**kwargs)
+            response_synthesizer = get_response_synthesizer(
+                response_mode=ResponseMode.COMPACT
```

Comment:
Now that we are excplitily defining a response synthesizer, should pass in the llm here, rather than the RetrieverQueryEngine

Reply:
sorry I'm confused about this one. Are you saying get_response_synthesizer() accepts an llm parameter?

Reply:
hmmm... when I move the LLM from RetrieverQueryEngine.from_args() to get_response_synthesizer() it somehow does not work. I get the response to be the prompt, as if the LLM was not called. ANy ideas?

Reply:
Just to be clear, you did this?

```
             response_synthesizer = get_response_synthesizer(
                 response_mode=ResponseMode.COMPACT,
                 llm=llm
             )
             return RetrieverQueryEngine.from_args(
                 retriever=retriever,
                 response_synthesizer=response_synthesizer,
                 **kwargs,
             )
```

If you don't include the llm in `get_response_synthesizer()` it will default to openai

`I get the response to be the prompt`  -- Is this because you were passing in `MockLLM` ?

Reply:
I get this msg: LLM is explicitly disabled. Using MockLLM.
Which probably is a hint of why this occurs. Any thoughts on why it won't recognize the llm?

Reply:
To be clear, you would need to do `as_query_engine(llm=llm)` to pass in the llm, or it will use the llm from `Settings`